controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -490,7 +599,7 @@ public String getScopeName() {
 
     @Override
     public CompletableFuture<TxnStatus> commitTransaction(final int epoch, final UUID txId) {
-        CompletableFuture<TxnStatus> future = verifyLegalState(checkTransactionStatus(epoch, txId).thenApply(x -> {
+        CompletableFuture<TxnStatus> future = verifyLegalState(() -> checkTransactionStatus(epoch, txId).thenApply(x -> {
```

Comment:
we do the blanket check of legal state for all get and update calls on the stream. 
legal state check merely checks if the stream is created or not.